### PR TITLE
[GIE-Runtime] Refine structure `Record` in Runtime

### LINKED ISSUE #1959: [GIE] Optimize `Record` in GAIA-Runtime
**Is your feature request related to a problem? Please describe.**

Optimize structure of `Record` in GAIA-Runtime. Now `Record` is a little bit complex and may involve  extra cost.

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

Refine structure `Record` by: 1). define a more flatten `Entry` in `Record`, 2) define label as `LabelId` rather than `NameOrId`, and 3) some other refinement. 
This brings about 30-50% acceleration in BI / pattern matching.

## Related issue number

Fixes #1959

### CODE REVIEW

File: interactive_engine/executor/ir/runtime/src/process/record.rs [hunk 1]
```diff
@@ -44,94 +42,54 @@ pub enum CommonObject {
...
-}
-
 #[derive(Debug, Clone, Hash, PartialEq, PartialOrd)]
 pub enum Entry {
-    Element(RecordElement),
-    Collection(Vec<RecordElement>),
+    V(Vertex),
+    E(Edge),
+    P(GraphPath),
+    OffGraph(CommonObject),
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/record.rs:50]: 不需要CommonObject，直接用Object如何？

File: interactive_engine/executor/ir/runtime/src/process/record.rs [hunk 2]
```diff
@@ -44,94 +42,54 @@ pub enum CommonObject {
...
+    Collection(Vec<Entry>),
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/record.rs:51]: `Collection`，具体放的是啥呀？为啥里面需要是一个Entry？

Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/record.rs:51]: `Collecrtion` 是比如 `fold()`，就会生成一个`Collection`的类型

File: interactive_engine/executor/ir/runtime/src/process/record.rs (hunk 1 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/record.rs]: Done.

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs [hunk 3]
```diff
@@ -127,19 +121,21 @@ impl GraphElement for VertexOrEdge {
         }
     }
 
-    fn label(&self) -> Option<&NameOrId> {
+    fn label(&self) -> Option<&LabelId> {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs:124]: 这个clone出来好了，不需要引用。后面的代码相应改下好了。

File: interactive_engine/executor/ir/graph_proxy/src/apis/read_graph.rs [hunk 4]
```diff
@@ -53,7 +53,7 @@ pub trait ReadGraph: Send + Sync {
     /// Scan a vertex with a specified label and its primary key value(s), and additional query parameters,
     /// and return the vertex if exists.
     fn index_scan_vertex(
-        &self, label: &NameOrId, primary_key: &PKV, params: &QueryParams,
+        &self, label: &LabelId, primary_key: &PKV, params: &QueryParams,
```
Comment (longbinlai) [inline interactive_engine/executor/ir/graph_proxy/src/apis/read_graph.rs:56]: 这个也不需要引用哈。Primitive type引用没有帮助

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs (hunk 3 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/path.rs]: Done

File: interactive_engine/executor/ir/graph_proxy/src/apis/read_graph.rs (hunk 4 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/graph_proxy/src/apis/read_graph.rs]: Done

File: interactive_engine/executor/ir/runtime/src/process/record.rs [hunk 5]
```diff
@@ -31,107 +29,54 @@ use pegasus::api::function::DynIter;
...
     pub fn as_graph_path(&self) -> Option<&GraphPath> {
         match self {
-            Entry::Element(record_element) => record_element.as_graph_path(),
+            Entry::P(p) => Some(p),
             _ => None,
         }
     }
 
-    pub fn as_common_object(&self) -> Option<&CommonObject> {
+    pub fn as_common_object(&self) -> Option<&Object> {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/record.rs:63]: `as_object()`

File: interactive_engine/executor/ir/runtime/src/process/record.rs [hunk 6]
```diff
@@ -31,107 +29,54 @@ use pegasus::api::function::DynIter;
...
         match self {
-            Entry::Element(record_element) => record_element.as_common_object(),
+            Entry::OffGraph(common_object) => Some(common_object),
             _ => None,
         }
     }
 
     pub fn as_mut_graph_path(&mut self) -> Option<&mut GraphPath> {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/record.rs:70]: `as_graph_path_mut()`

File: interactive_engine/executor/ir/runtime/src/process/record.rs (hunk 5 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/record.rs]: Done

File: interactive_engine/executor/ir/runtime/src/process/record.rs (hunk 6 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/record.rs]: Done

**Review by longbinlai — APPROVED** [review]

**Pull request merged**